src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/Items.java

```diff
@@ -455,7 +455,7 @@ void store() {
         }
 
         Item invoke() {
-            MethodType mtype = (MethodType)member.erasure(types);
+            MethodType mtype = (MethodType)member.externalType(types);
```

Comment:
not sure about this change

Reply:
The thing is that when the time comes to Generate a var def. For example `/*synthetic*/ final Object patt4$temp = ExampleStringTest$StringOps$Split.Split($b$0)`, the result needs to be `stack(object)` instead of `void` for the deconstructor call. The method type of split needs to be `(ExampleStringTest.StringOps.Split)java.lang.Object`. How can I change that earlier?